src/main/java/nextstep/auth/application/TokenService.java

```diff
@@ -1,23 +1,25 @@
-package nextstep.member.application;
+package nextstep.auth.application;
 
-import nextstep.member.AuthenticationException;
-import nextstep.member.application.dto.GithubProfileResponse;
+import nextstep.auth.AuthenticationException;
+import nextstep.auth.application.dto.GithubProfileResponse;
+import nextstep.auth.domain.OAuthProvider;
+import nextstep.auth.domain.OAuthUser;
+import nextstep.member.application.MemberService;
 import nextstep.member.application.dto.MemberResponse;
-import nextstep.member.application.dto.TokenResponse;
-import nextstep.member.domain.GithubClient;
+import nextstep.auth.application.dto.TokenResponse;
 import nextstep.member.domain.Member;
 import org.springframework.stereotype.Service;
 
 @Service
 public class TokenService {
     private MemberService memberService;
     private JwtTokenProvider jwtTokenProvider;
-    private GithubClient githubClient;
+    private GithubOAuthService githubOAuthService;
 
```

Comment:
현재는 oAuth를 제공해주는 주체가 Github 밖에 없지만 우석님이 `OauthProvider` Enum 에서 선언해주신 것처럼 '카카오, 네이버의 OAuth를 추가로 제공'해야 하는 상황이 올 수 있을거에요.
그런 상황에 유연하게 대처하기 위해 추상화를 잘 구현해주셨고, 구현한 내용을 활용해서 TokenSerivce에서  아래와 같이 빈을 주입 받는다면 추상화한 목적을 더 명확히 해볼 수 있을 것 같습니다! (참고 : [Spring - Injecting collections](https://www.baeldung.com/spring-injecting-collections))
```suggestion
    private Map<OAuthProvider, OAuthService> githubOAuthService;
```

Reply:
Map을 사용하면 enum값을 기준으로 switch를 돌릴 필요가 없겠네요! 좋은 방법을 알려주셔서 감사합니다 😁